### PR TITLE
Update PeripheralPins to SERCOM4 default

### PR DESCRIPTION
**Pull request by sabas1080**
Hi

Fix bug of mixture SERCOM2 with SERCOM4

Pull request to work module Lora (SPI) on MKR 1300

Thanks

### CODE REVIEW

File: targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c [hunk 1]
```diff
@@ -58,9 +58,9 @@ const PinMap PinMap_SERCOM_PAD[] = {
     {PA10, SERCOM0_PAD2, 2},
     {PA11, SERCOM0_PAD3, 2},
     {PA12, SERCOM4_PAD0, 3},
```
Comment (mmoskal) [inline targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c:60]: @finneyj @jamesadevine note that PA12-15 should be all on the same SERCOM, and they were not

File: targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c [hunk 2]
```diff
@@ -75,10 +75,10 @@ const PinMap PinMap_SERCOM_PAD[] = {
     {PA31, SERCOM1_PAD3, 3},
     {PB02, SERCOM5_PAD0, 3},
     {PB03, SERCOM5_PAD1, 3},
-    {PB08, SERCOM4_PAD0, 3},
-    {PB09, SERCOM4_PAD1, 3},
-    {PB10, SERCOM4_PAD2, 3},
-    {PB11, SERCOM4_PAD3, 3},
+    {PB08, SERCOM2_PAD0, 2},
```
Comment (jamesadevine) [inline targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c:78]: @sabas1080 I'm not sure these changes to PB08 - PB11 are correct. Can you confirm you meant to include these in your PR?

Comment (jamesadevine) [reply targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c:78]: (feel free to tell me why I'm wrong 😄 )

Comment (sabas1080) [reply targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A/PeripheralPins.c:78]: when the two of them are in SERCOM4 there is conflict

**Comment by pelikhan** [comment]
ping

**Comment by jamesadevine** [comment]
@pelikhan @sabas1080 

I'm reluctant to merge this as this will break a flag ship board ([the CPX](https://learn.adafruit.com/assets/49671)). You can point your target.json to point at your fork with the above patch if required. 

Thanks for pushing your patch back @sabas1080, I hope you understand my reasoning.

**Comment by sabas1080** [comment]
@jamesadevine  Why stop working? is mixing two sercom, the current configuration is incorrect, have you already tried this configuration in CPX of @adafruit ?

**Comment by jamesadevine** [comment]
Hi @sabas1080,

You raise a good point – I only read the datasheet.

I therefore tested it this morning and it does indeed break serial on the circuit playground. I have attached the two test files, one built using your PR, the other built on the current branch `new-build-system`.

[test-uf2-files.zip](https://github.com/lancaster-university/mbed-classic/files/2391927/test-uf2-files.zip)

**Comment by jamesadevine** [comment]
@sabas1080 can you give a little bit more information on what you're working on? There are a number of ways to achieve what you want.

What device are you working on? Is it for fun or a product?

**Comment by sabas1080** [comment]
@jamesadevine 

I'm trying to test the changes in dafruit makecode, but in doing so I can not finish compiling. with another version of makecode I have no error

https://makecode.com/compile/dc2c625a65ecf198e708fad77c86a133225004f050b89fc3f0b560c32dfcad03.log

**Comment by sabas1080** [comment]
@jamesadevine  I am working on a picosatelite with MKR1300, is an product

https://github.com/ElectronicCats/CatSatZero

communication via lora, sensors BMP280, CCS811 and GPS, "the first picosatelite programmed by kids" 😅

**Comment by jamesadevine** [comment]
@sabas1080 Ok, so you're developing an accessory for the circuit playground and would like to have it also available from MakeCode? 

Can you help me to understand why this patch is required? I see that the lora module appears to use UART....

**Comment by sabas1080** [comment]
I'm developing an accessory for the MKR1300 of arduino and would like to have MKR1300 i also available from MakeCode 

- Can you help me to understand why this patch is required? 
UART read function not working in makecode, I can not know the answer of the olive chip.
But SPI works perfectly, I have 80% of package of lora with SPI

**Comment by jamesadevine** [comment]
@sabas1080 Ok – so the way we add new boards to mbed is a little bit different to the way you approached it. 

We like to have separation between targets so that one can be updated without the other being affected, a prime example being pin maps :smile:

https://github.com/lancaster-university/mbed-classic/tree/new-build-system/targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_ADAFRUIT_CP_G18A

You can see we have a folder for the pins of the adafruit circuit playground. The way this is "found" during compile time is through mbed's use of defines, which you can see here:

https://github.com/lancaster-university/codal-circuit-playground/blob/master/target.json#L41

These get transformed into a path that eventually routes to the folder above. 

My suggestion would be to create your own target that looks like `codal-circuit-playground` called `codal-mkr1300`, that modifies the target.json to point at a different folder with your custom pin mappings.  This folder would be added [here](https://github.com/lancaster-university/mbed-classic/tree/new-build-system/targets/hal/TARGET_Atmel/TARGET_SAM_CortexM0P) and called something like TARGET_ARDUINO_MKR1300.

**Comment by sabas1080** [comment]
Ok,

> We like to have separation between targets so that one can be updated without the other being affected, a prime example being pin maps 😄

I have added a pull request with the changes

> My suggestion would be to create your own target that looks like `codal-circuit-playground` called `codal-mkr1300`

Is this to open a repository with this name?

Thanks

**Comment by jamesadevine** [comment]
Give me an hour or two and I’ll set that up for you and give you a blink to try :)


James.

On 20 Sep 2018, at 16:58, Sabas <notifications@github.com<mailto:notifications@github.com>> wrote:


Ok,

We like to have separation between targets so that one can be updated without the other being affected, a prime example being pin maps 😄

I have added a pull request with the changes

My suggestion would be to create your own target that looks like codal-circuit-playground called codal-mkr1300

Is this to open a repository with this name?

Thanks

**Comment by mmoskal** [comment]
Note that in PXT we've been using the CPX target for all SAMD21 boards. It can be changed but requires some setup.

**Comment by jamesadevine** [comment]
@sabas1080 could you let me know if this does anything? I have the led as PA20, correct?

[mkr1300-bins.zip](https://github.com/lancaster-university/mbed-classic/files/2402283/mkr1300-bins.zip)

**Comment by sabas1080** [comment]
not blink, yes led in pin PA20

**Comment by sabas1080** [comment]
@mmoskal How would that configuration be? Can CPX and MKR coexist?

**Comment by sabas1080** [comment]
@jamesadevine @mmoskal  I create a fork that works well SPI and blink 

https://github.com/ElectronicCats/codal-circuit-playground

based in snapshot 1.4.4 is snapshot 1.4.5-mixer

**Comment by jamesadevine** [comment]
That's great news @sabas1080. I think you'll have to run your own makecode instance (presumably your doing that already?).

I created the [MKR1300 target](https://github.com/lancaster-university/codal-mkr1300), and added the new target to [codal](https://github.com/lancaster-university/codal/commit/1c813c0f4698c65c9995333a81cff8ec533be2c8). This is the source from which you said the blinky didn't work!

**Comment by sabas1080** [comment]
@jamesadevine I can not compile after version [1.4.4](https://github.com/lancaster-university/codal-circuit-playground/releases/tag/v1.4.4) of `codal circuit playground` and I see that `codal-mkr1300` is based on a later version, if you modify this version I'm sure it will work, is the version for pxt-maker https://github.com/Microsoft/pxt-maker/blob/master/pxtarget.json#L113

**Pull request closed without merging**